packages/@okta/vuepress-site/docs/guides/terraform-syntax-tips/main/index.md

```diff
@@ -0,0 +1,199 @@
+
+```hcl
+variable "groups_to_create" {
+  description = "List of group objects"
+  type = list(object({
+    name = string
+    description = string
+   }))
+}
+```
+
+2. Create a file for variable values, in this case, a list of group objects to create. If you aren't using the central `terraform.tfvars` file to set variable values, name the file with the suffix `.auto.tfvars`. For example, `users_and_groups.auto.tfvars`.
+
+Add this example to the file:
+
+```hcl
+groups_to_create = [
+  {
+    "name": "Group 1",
+    "description": "Group 1 description"
+  },
+  {
+    "name": "Group 2",
+    "description": "Group 2 description"
+  },
+  {
+    "name": "Group 3",
+    "description": "Group 3 description"
+  }
+]
+```
+
+3. In your original Terraform `.tf` file, create a group resource definition:
+
+```hcl
+resource "okta_group" "group_create_using_list" {
+
+}
+```
+
+4. In that resource definition, set `for_each` to a map or set. You can convert array content to a map before setting the meta-argument:
+
+```hcl
+for_each = { for group in var.&lt;your-variable-name&gt;
+  : group.name =&gt; group}
+```
+
+Using the previous examples:
+
+```hcl
+for_each = { for group in var.groups_to_create
+  : group.name =&gt; group}
+```
+
+5. Add additional lines where you use `each.value` to access the current group in the loop.
+
+Example of the resource definition that creates one group for each member of the original list:
+
+```hcl
+resource "okta_group" "group_create_using_list" {
+
+  # Convert the array to a map
+    for_each = { for group in var.groups_to_create
+    : group.name =&gt; group}
+   # Use `each.value` to access each member of your list
+  name        = each.value.name
+  description = each.value.description
+}
+```
+
+### Avoid rate limit errors from loops
+
+Remember that Terraform uses the Okta Management APIs when you design loops. Follow the guidance in the related article [optimizing your Terraform deployment to avoid hitting rate limits](/docs/guides/terraform-design-rate-limits/main/).
+
+For example, the following loop iterates through the users in the Everyone group. This results in iterating through all users in the system, something that's likely to result in a rate limit error:
+
+```hcl
+# Assign every user to an app
+data "okta_group" "everyone_group" {
+  name = "Everyone"
+  include_users = "true"
+}
+
+data "okta_app" "test_application" {
+  label = "Test Application"
+}
+
+resource "okta_app_user" "by_resource" {
+  for_each = { for user_ref in data.okta_group.everyone_group.users : user_ref =&gt; user_ref }
+  app_id = data.okta_app.test_application.id
+  user_id = each.value
+}
+```
+
+Instead, Okta recommends managing objects, such as apps, using groups rather than users. This version of the code uses a group rule. It runs faster and has a lower risk of resulting in an API rate limit error:
```

Comment:
There's a problem here. "This version of the code uses a group rule" implies that this is an improved version of the previous example but I'm not sure it's doing the same thing. 

I think the previous example iterates over all the users in the Everyone group and assigns them to an app called test_application. (Admittedly I'm not sure)
I think this example assigns a group called employee_group to an app called okta_terraform_app. I'm not even sure what the group rule is or how it is relevant. Apparently it is very relevant but you'd need to explain that in text and how it is equivalent.
I was expecting to see an example that simply iterated over a list of groups and applied them to the app instead of a list of users.

Reply:
I have to assume it's syntactically and semantically correct... it's gone through a tech review by the folks who write the Terraform Provider. I'll leave it as is and await customer feedback... you know: "Just get it our there and fix it later" ;-).